src/general/OptionsMenu.cs

```diff
@@ -2280,8 +2280,14 @@ private void OnJSONDebugModeSelected(int index)
 
     private void OnScreenEffectSelected(int index)
     {
-        Settings.Instance.CurrentScreenEffect.Value =
-            SimulationParameters.Instance.GetScreenEffectByIndex(index);
+        var currentEffect = Settings.Instance.CurrentScreenEffect;
```

Comment:
Rather than getting the effect value like this and doing things the unintended way. I'd suggest rewriting this like:

```
var effect = SimulationParameters.Instance.GetScreenEffectByIndex(index);

if (effect == SimulationParameters.Instance.GetScreenEffectByIndex(0))
            effect = null;

Settings.Instance.CurrentScreenEffect.Value = effect;
```

Because assigning the effect's value back to itself like this: `Settings.Instance.CurrentScreenEffect.Value = currentEffect.Value;` doesn't do anything. Or if it does it is actually a bug in the settings system.